diff --git a/git-request-pull.sh b/git-request-pull.sh
index d566015..71abbf4 100755
--- a/git-request-pull.sh
+++ b/git-request-pull.sh
@@ -79,7 +79,7 @@ find_matching_ref='
 		my ($sha1, $ref, $deref) = /^(\S+)\s+(\S+?)(\^\{\})?$/;
 		next unless ($sha1 eq $ARGV[1]);
 		$found = abbr($ref);
-		if ($deref && $ref eq "tags/$ARGV[2]") {
+		if ($deref && $ref eq "refs/tags/$ARGV[2]") {
 			$tagged = $found;
 			last;
 		}
